posts/2021-11-02-authenticating-requests-using-saml.md

```diff
@@ -0,0 +1,45 @@
+* Amazon IAM
+* Amazon Cognito user pools
+* Custom authorizers using AWS Lambda
+
+In practice, one node can use Amazon IAM, another could use an Amazon Cognito user pool, and another could use a custom authorizer - all while sharing the same consistent and permissioned view of data. In this post, we're going to focus on Amazon Cognito user pools and, specifically, how to take advantage of an existing SAML 2.0 identity store to authenticate API requests to Vendia Share.
+
+We'll begin by looking at SAML and the value it affords application end-users and identity administrators.
+
+# SAML
+
+> Security Assertion Markup Language 2.0 (SAML) is an open federation standard that allows an identity provider (IdP) to authenticate users and pass identity and security information about them to a service provider (SP), typically an application or service. With SAML, you can enable a single sign-on experience for your users across many SAML-enabled applications and services. Users authenticate with the IdP once using a single set of credentials, and then get access to multiple applications and services without additional sign-ins. Because SAML-enabled applications delegate authentication to an IdP, the SP can automatically grant, revoke, or change the scope of a user’s access to applications and services when an administrator adds, removes, or modifies the user’s information in the IdP.
+> - AWS Identity (https://aws.amazon.com/identity/saml/)
+
+In short, SAML allows users to authenticate to an existing identity provider and use generated credentials to access a downstream service or application, thus enabling a single sign-on experience. End users have the benefit of not having to use yet another set of credentials when accessing applications or services. Administrators improve their ability to manage users across a fleet of services. Providing and revoking access to SAML-enabled services becomes a streamlined process, all from one source - _the IdP_.
+
+Now that we have a high-level understanding of SAML and the value it provides, let's explore Cognito user pools and how they can work together with SAML and Vendia Share.
+
+# SAML + Cognito User Pools + Vendia Share
+
+A user pool is a user directory in Amazon Cognito. It allows users to sign in to web and mobile applications through Amazon Cognito. User pools can be configured to allow users to sign in through SAML IdPs.
+
+Many identity services, including Okta, Ping Identity, Auth0, and Microsoft Active Directory Federation Services (ADFS), can function as SAML IdPs and work seamlessly with Cognito user pools. Regardless of the IdP, Amazon Cognito will return a JWT token to your mobile and web applications after a successful login. [AWS has created the following diagram to illustrate the authentication flow](https://docs.aws.amazon.com/cognito/latest/developerguide/images/amazon-cognito-user-pools-saml-idp-auth-flow.png).
```

Comment:
Maybe links to each of the IdPs you mentioned for ease of nav?

Reply:
Good call - updated in latest push.